src/Components/Web.JS/src/Services/WebRootComponentManager.ts

```diff
@@ -354,67 +358,52 @@ export class WebRootComponentManager implements DescriptorHandler, RootComponent
       this.unregisterComponent(component);
-      if (component.assignedRendererId !== undefined && component.interactiveComponentId !== undefined) {
-        const renderer = getRendererer(component.assignedRendererId);
-        renderer?.disposeComponent(component.interactiveComponentId);
-      }
 
-      if (component.interactiveComponentId !== undefined) {
-        // We have an interactive component for this marker, so we'll remove it.
-        return { type: 'remove', componentId: component.interactiveComponentId };
+      if (component.assignedRendererId === undefined) {
+        // The component was removed from the document before it was assigned to a renderer,
+        // so we don't have to notify .NET that anything has changed.
+        return null;
       }
 
-      // If we make it here, that means we either:
-      // 1. Haven't started to make the component interactive, in which case we have no further action to take.
-      // 2. Have started to make the component interactive, but it hasn't become interactive yet. In this case,
-      //    we'll wait to remove the component until after we have a component ID to provide.
-    }
+      // Since the component will be getting completedly diposed from .NET (rather than replaced by another component, which can
+      // happen as a result of an 'update' operation), we indicate that its content should no longer be preserved on disposal.
+      setShouldPreserveContentOnInteractiveComponentDisposal(component.descriptor.start as unknown as LogicalElement, false);
 
-    return null;
+      // This component was removed from the document and we've assigned a renderer ID,
+      // so we'll dispose it in .NET.
+      return { type: 'remove', ssrComponentId: component.ssrComponentId };
+    }
   }
 
-  public resolveRootComponent(selectorId: number, componentId: number): ComponentDescriptor {
-    const component = this._pendingComponentsToResolve.get(selectorId);
+  public resolveRootComponent(ssrComponentId: number): ComponentDescriptor {
+    const component = this._rootComponentsBySsrComponentId.get(ssrComponentId);
     if (!component) {
-      throw new Error(`Could not resolve a root component for descriptor with ID '${selectorId}'.`);
+      throw new Error(`Could not resolve a root component with SSR component ID '${ssrComponentId}'.`);
     }
 
-    this._pendingComponentsToResolve.delete(selectorId);
-
-    if (component.interactiveComponentId !== undefined) {
-      throw new Error('Cannot resolve a root component for the same descriptor multiple times.');
-    }
-
-    component.interactiveComponentId = componentId;
-
-    // The descriptor may have changed since the last call to handleUpdatedRootComponentsCore().
-    // We'll update this single descriptor so that the component receives the most up-to-date parameters
-    // or gets removed if it no longer exists on the page.
-    this.refreshRootComponents([component]);
-
```

Comment:
These are the types of things we no longer need to worry about by allowing .NET to control the mapping between SSR'd components and interactive components.

Reply:
Impressive